src/python/library/tritonclient/utils/shared_memory/shared_memory.cc

```diff
@@ -23,34 +23,172 @@
+}
+
+}  // namespace
+
+TRITONCLIENT_DECLSPEC int
+SharedMemoryRegionCreate(
+    const char* triton_shm_name, const char* shm_key, size_t byte_size,
+    void** shm_handle)
+{
+  // The CreateFileMapping function takes a high-order and low-order DWORD (4
+  // bytes each) for size. 'size_t' can either be 4 or 8 bytes depending on the
+  // operating system. To handle both cases agnostically, we cast 'byte_size' to
+  // uint64 to ensure we have a known size and enough space to perform our
+  // logical operations.
+  uint64_t upperbound_size = (uint64_t)byte_size;
+  DWORD high_order_size = (upperbound_size >> 32) & 0xFFFFFFFF;
+  DWORD low_order_size = upperbound_size & 0xFFFFFFFF;
+
+  HANDLE local_handle = CreateFileMapping(
+      INVALID_HANDLE_VALUE,  // use paging file
+      NULL,                  // default security
+      PAGE_READWRITE,        // read/write access
+      high_order_size,       // maximum object size (high-order DWORD)
+      low_order_size,        // maximum object size (low-order DWORD)
+      shm_key);              // name of mapping object
+
+  if (local_handle == NULL) {
+    return -7;
+  }
+
+  // get base address of shared memory region
+  void* shm_addr = nullptr;
+  int err = SharedMemoryRegionMap(local_handle, 0, byte_size, &shm_addr);
+  if (err == -1) {
+    return -4;
+  }
+
+  // create a handle for the shared memory region
+  *shm_handle = SharedMemoryHandleCreate(
+      std::string(triton_shm_name), shm_addr, std::string(shm_key),
+      local_handle, 0, byte_size);
+  return 0;
+}
+
+TRITONCLIENT_DECLSPEC int
+SharedMemoryRegionSet(
+    void* shm_handle, size_t offset, size_t byte_size, const void* data)
+{
+  void* shm_addr =
+      reinterpret_cast<SharedMemoryHandle*>(shm_handle)->base_addr_;
+  char* shm_addr_offset = reinterpret_cast<char*>(shm_addr);
+  std::memcpy(shm_addr_offset + offset, data, byte_size);
+  return 0;
+}
+
+TRITONCLIENT_DECLSPEC int
+GetSharedMemoryHandleInfo(
+    void* shm_handle, char** shm_addr, const char** shm_key,
+    SHM_FILE* shm_file_handle, size_t* offset, size_t* byte_size)
+{
+  SharedMemoryHandle* handle =
+      reinterpret_cast<SharedMemoryHandle*>(shm_handle);
+  *shm_addr = reinterpret_cast<char*>(handle->base_addr_);
+  *shm_key = handle->shm_key_.c_str();
+  *shm_file_handle = handle->shm_handle_;
+  *offset = handle->offset_;
+  *byte_size = handle->byte_size_;
+  return 0;
+}
+
+TRITONCLIENT_DECLSPEC int
+SharedMemoryRegionDestroy(void* shm_handle)
+{
+  SharedMemoryHandle* handle =
+      reinterpret_cast<SharedMemoryHandle*>(shm_handle);
+  void* shm_addr = reinterpret_cast<char*>(handle->base_addr_);
+  bool success = UnmapViewOfFile(shm_addr);
+  if (!success) {
+    return -6;
+  }
+  // We keep Windows shared memory handles open until we are done
+  // using them. When all handles are closed, the system will free
+  // the section of the pagaing file that the object uses.
+  CloseHandle(handle->shm_handle_);
+
+  return 0;
+}
+#else
```

Comment:
Yes, I was torn on whether to separate the logic or integrate the logic. Ultimately, I decided to separate the logic for two reasons:
1. We followed this approach in [shared_memory_manager.cc](https://github.com/triton-inference-server/server/blob/main/src/shared_memory_manager.cc#L30-L119).
2. I wanted to keep the Windows/Linux implementations as separate as possible until we'd built up confidence with the Windows port and felt like it would be easier to identify the Windows from Linux implementation with the current changes.

However, I will integrate the changes as much as a can given this feedback.